src/cbor.c

```diff
@@ -140,6 +140,31 @@ int cbor_serialize_string(uint8_t *data, const char *value, size_t max_len)
     }
 }
 
+int cbor_serialize_bytes(uint8_t *data, const uint8_t *bytes, size_t num_bytes, size_t max_len)
+{
+    if (num_bytes < 24 && num_bytes + 1 <= max_len) {
+        data[0] = CBOR_BYTES | (uint8_t)num_bytes;
+        memcpy(&data[1], bytes, num_bytes);
+        return num_bytes + 1;
+    }
+    else if (num_bytes < 0xFF && num_bytes + 2 <= max_len) {
+        data[0] = CBOR_BYTES | CBOR_UINT8_FOLLOWS;
+        data[1] = (uint8_t)num_bytes;
+        memcpy((char*)&data[2], bytes, num_bytes);
+        return num_bytes + 2;
+    }
+    else if (num_bytes < 0xFFFF && num_bytes + 3 <= max_len) {
```

Comment:
is `0xFFFF` meant as MAX_UINT16? maybe there is also a label?

Reply:
Good spot. Will fix that.